assets/shared/helpers/player.js

```diff
@@ -0,0 +1,557 @@
+		 *
+		 * @param {Object} player The YouTube player instance.
+		 *
+		 * @return {Promise} A resolved promise.
+		 */
+		pause: ( player ) => {
+			player.pauseVideo();
+			return Promise.resolve();
+		},
+
+		/**
+		 * Add an event listener to the player.
+		 *
+		 * @param {Object}   player    The YouTube player instance.
+		 * @param {string}   eventName Event name (currently only `timeupdate` is supported)
+		 * @param {Function} callback  Listener callback.
+		 *
+		 * @return {Function} The function to unsubscribe the event.
+		 */
+		on: ( player, eventName, callback ) => {
+			const timer = 250;
+
+			const interval = setInterval( () => {
+				if (
+					player.getPlayerState() === window.YT.PlayerState.PLAYING
+				) {
+					callback( player.getCurrentTime() );
+				}
+			}, timer );
+
+			return () => {
+				clearInterval( interval );
+			};
+		},
+	},
+	[ VIMEO_TYPE ]: {
+		/**
+		 * The embed pattern to check if it's the respective type.
+		 */
+		embedPattern: /vimeo\.com\/.+/i,
+
+		/**
+		 * Initialize the player.
+		 *
+		 * @param {HTMLIFrameElement} element The player element.
+		 *
+		 * @return {Object} The Vimeo player instance through a promise.
+		 */
+		initializePlayer: ( element ) =>
+			Promise.resolve( new window.Vimeo.Player( element ) ),
+
+		/**
+		 * Get the video duration.
+		 *
+		 * @param {Object} player The Vimeo player instance.
+		 *
+		 * @return {Promise<number>} The duration of the video in seconds through a promise
+		 *                           (original return from Vimeo API).
+		 */
+		getDuration: ( player ) => player.getDuration(),
+
+		/**
+		 * Set the video to a current time.
+		 *
+		 * @param {Object} player  The Vimeo player instance.
+		 * @param {number} seconds The video time in seconds to set.
+		 *
+		 * @return {Promise} A promise that resolves if the video was set to a current time successfully.
+		 *                   (original return from Vimeo API).
+		 */
+		setCurrentTime: ( player, seconds ) => player.setCurrentTime( seconds ),
+
+		/**
+		 * Play the video.
+		 *
+		 * @param {Object} player The Vimeo player instance.
+		 *
+		 * @return {Promise} A promise that resolves if the video was played successfully.
+		 *                   (original return from Vimeo API).
+		 */
+		play: ( player ) => player.play(),
+
+		/**
+		 * Pause the video.
+		 *
+		 * @param {Object} player The Vimeo player instance.
+		 *
+		 * @return {Promise} A promise that resolves if the video was paused successfully.
+		 *                   (original return from Vimeo API).
+		 */
+		pause: ( player ) => player.pause(),
+
+		/**
+		 * Add an event listener to the player.
+		 *
+		 * @param {Object}   player    The Vimeo player instance.
+		 * @param {string}   eventName Event name (currently only `timeupdate` is supported)
+		 * @param {Function} callback  Listener callback.
+		 *
+		 * @return {Function} The function to unsubscribe the event.
+		 */
+		on: ( player, eventName, callback ) => {
```

Comment:
Similar comment here re: checking for `timeupdate`.

Reply:
https://github.com/Automattic/sensei/pull/5364#discussion_r929259028